### PR TITLE
Fix docstrings, enhance docs and change an assertion

### PR DESCRIPTION
**Pull request by thatbrguy**
### Description
- Fixed docstring in `projections_from_fundamental` so that #996 and #997 can be closed.
- Fixed docstrings in `eye_like` and `vec_like`. 
- Changed an assertion in the function `projections_from_fundamental` so that the shape is checked properly (that's why I selected breaking change in the types of changes below. Do let me know if this is not considered a breaking change).
- Enhanced some documentation.

I ran the following tests:
1. `pytest -sv test/geometry/epipolar/test_projection.py  --device all --dtype float32`: All passed with 1 warning (explained below)
2. `pytest -sv test/geometry/epipolar/test_projection.py  --device all --dtype float64`: All passed with 1 warning (explained below)
3. `pytest -sv test/geometry/epipolar/test_numeric.py  --device all --dtype float32`: All passed
4. `pytest -sv test/geometry/epipolar/test_numeric.py  --device all --dtype float64`: All passed
5. `make build-docs`: Build succeeded with 2 warnings (the usual name 'np' is not defined warnings)

The warning obtained for `test_projection.py` for both dtypes tested were similar. The warnings summary obtained when dtype was `float64` was:

```
=================================================================== warnings summary ====================================================================
test/geometry/epipolar/test_projection.py::TestProjectionFromKRt::test_krt_from_projection[cpu-float64]
  /kornia/kornia/geometry/epipolar/projection.py:134: UserWarning: torch.qr is deprecated in favor of torch.linalg.qr and will be removed in a future PyTorch release.
  The boolean parameter 'some' has been replaced with a string parameter 'mode'.
  Q, R = torch.qr(A, some)
  should be replaced with
  Q, R = torch.linalg.qr(A, 'reduced' if some else 'complete') (Triggered internally at  /opt/conda/conda-bld/pytorch_1623448224956/work/aten/src/ATen/native/BatchLinearAlgebra.cpp:1940.)
    ortho_mat, upper_mat = torch.qr(submat_3x3)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
```

Perhaps someone can address the warning in a separate PR later. 

Also, I got some fails in `make test` since I kept running out of CUDA memory (my laptop has only 2GB GPU memory). It's odd cause when I was submitting an older PR I was able to run `make test` just fine without the above memory issue. Not sure why it's running out of memory now (maybe because of any changes/additions to tests since last PR or maybe because of some issue in my laptop). Have to check running `make test` with more GPU memory somewhere else sometime. Anyway, just wanted to mention this.

Do let me know if I have to make any changes to this PR. Thanks!

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [x] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>

  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?

</details>

### CODE REVIEW

File: kornia/geometry/epipolar/numeric.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def eye_like(n: int, input: torch.Tensor) -> torch.Tensor:
           The expected shape is :math:`(B, *)`.
 
     Returns:
-       The identity matrix with same size as input :math:`(*, N, N)`.
+       The identity matrix with same size as input :math:`(B, N, N)`.
```
Comment (edgarriba) [inline kornia/geometry/epipolar/numeric.py:40]: unrelated to this PR but do you think that we should move `eye_like` and `vec_like` under `kornia.utils ? and expose to top level. /cc @ducha-aiki @shijianjian

**Review by edgarriba — APPROVED** [review]
possibly for a next PR too - add small numerical test in https://github.com/kornia/kornia/blob/master/test/geometry/epipolar/test_projection.py#L121

Regarding the QR thing - we need something similar to `solve` to cast and keep backward compatibility. See: https://github.com/kornia/kornia/blob/master/kornia/utils/_compat.py#L14

**Comment by thatbrguy** [comment]
> possibly for a next PR too - add small numerical test in https://github.com/kornia/kornia/blob/master/test/geometry/epipolar/test_projection.py#L121
> 
> Regarding the QR thing - we need something similar to `solve` to cast and keep backward compatibility. See: https://github.com/kornia/kornia/blob/master/kornia/utils/_compat.py#L14

Hmm adding a numerical test is a nice idea. Do you have any suggestions for the same ? I may have to research a bit about it also. Maybe we could discuss further on slack. 

Yeah I guess the QR thing could be handled in the way you suggested. Perhaps an issue could be opened for the time being.

**Comment by edgarriba** [comment]
@thatbrguy for numerical tests I would just hardcode some numbers just to make sure that each time we keep same results.

For the QR thing, please go ahead and open an issue.

File: kornia/geometry/epipolar/numeric.py (hunk 1 above)
Comment (thatbrguy) [reply kornia/geometry/epipolar/numeric.py:40]: I think those functions are already accessible from `kornia.eye_like` and `kornia.vec_like` respectively. But yeah I guess they can be moved to a more generic module like `konria.utils`

**Pull request merged**